doc/doxygen/headers/automatic_and_symbolic_differentiation.h

```diff
@@ -20,8 +20,34 @@
  * to automatic and symbolic differentiation.
  *
  * Below we provide a very brief introduction as to what automatic and symbolic differentiation are,
- * what variations of these computational / numerical schemes exist, and how they are integrated
- * within deal.II's framework.
+ * what variations of these computational/numerical schemes exist, and how they are integrated
+ * within deal.II's framework. The purpose of all of these schemes is to automatically compute
+ * (approximations of) the derivative of functions in cases where one does not want to
+ * compute them by hand. Common examples are situations in the finite element context is where
+ * one wants to solve a nonlinear problem that is given by requiring that some residual
+ * $F(u,\nabla u)=0$ where $F$ is a complicated function that needs to be differentiated to
+ * apply a Newton method; and situations where one is given a parameter dependent problem
```

Comment:
Maybe this should be "apply Newton's method" or "apply a Newton-like method". I suppose there is only one "Newton–Raphson method".